metadata-ingestion/src/datahub/ingestion/source/qlik_sense/qlik_sense.py

```diff
@@ -0,0 +1,529 @@
+
+    @staticmethod
+    def test_connection(config_dict: dict) -> TestConnectionReport:
+        test_report = TestConnectionReport()
+        try:
+            QlikAPI(QlikSourceConfig.parse_obj_allow_extras(config_dict))
+            test_report.basic_connectivity = CapabilityReport(capable=True)
+        except Exception as e:
+            test_report.basic_connectivity = CapabilityReport(
+                capable=False, failure_reason=str(e)
+            )
+        return test_report
+
+    @classmethod
+    def create(cls, config_dict, ctx):
+        config = QlikSourceConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def _gen_space_key(self, space_id: str) -> SpaceKey:
+        return SpaceKey(
+            space=space_id,
+            platform=self.platform,
+            instance=self.config.platform_instance,
+        )
+
+    def _gen_app_key(self, app_id: str) -> AppKey:
+        return AppKey(
+            app=app_id,
+            platform=self.platform,
+            instance=self.config.platform_instance,
+        )
+
+    def _get_audit_stamp(self, date: datetime, username: str) -> AuditStampClass:
+        return AuditStampClass(
+            time=int(date.timestamp() * 1000),
+            actor=builder.make_user_urn(username),
+        )
+
+    def _get_allowed_spaces(self) -> List[Space]:
+        all_spaces = self.qlik_api.get_spaces()
+        allowed_spaces = [
+            space
+            for space in all_spaces
+            if self.config.space_pattern.allowed(space.name)
+        ]
+        logger.info(f"Number of spaces = {len(all_spaces)}")
+        self.reporter.report_number_of_spaces(len(all_spaces))
+        logger.info(f"Number of allowed spaces = {len(allowed_spaces)}")
+        return allowed_spaces
+
+    def _gen_space_workunit(self, space: Space) -> Iterable[MetadataWorkUnit]:
+        """
+        Map Qlik space to Datahub container
+        """
+        yield from gen_containers(
+            container_key=self._gen_space_key(space.id),
+            name=space.name,
+            description=space.description,
+            sub_types=[BIContainerSubTypes.QLIK_SPACE],
+            extra_properties={Constant.TYPE: str(space.type)},
+            owner_urn=builder.make_user_urn(space.ownerId)
+            if self.config.ingest_owner and space.ownerId
+            else None,
+            created=int(space.createdAt.timestamp() * 1000),
+            last_modified=int(space.updatedAt.timestamp() * 1000),
+        )
+
+    def _gen_entity_status_aspect(self, entity_urn: str) -> MetadataWorkUnit:
+        return MetadataChangeProposalWrapper(
+            entityUrn=entity_urn, aspect=Status(removed=False)
+        ).as_workunit()
+
+    def _gen_entity_owner_aspect(
+        self, entity_urn: str, owner_id: str
+    ) -> MetadataWorkUnit:
+        aspect = OwnershipClass(
+            owners=[
+                OwnerClass(
+                    owner=builder.make_user_urn(owner_id),
```

Comment:
can we resolve these to actual users or emails or something? showing a random id doesn't really make sense

<img width="385" alt="image" src="https://github.com/datahub-project/datahub/assets/12566801/2139b9de-ffcb-4a6b-b558-c0bdbdd00d17">